File: lrpc-csharp/buf/SendData.cs
using lrpc.val;

namespace lrpc.buf
{
    /// <summary>
    /// add length to the actual data to judge the integrity of the data
    /// </summary>
    public class SendData
    {
        private ByteQue buff;

        /// <param name="que">the actual data</param>
        public SendData(ByteQue que)
        {
            buff = que;
        }

        /// <summary>
        /// added length data
        /// </summary>
        public byte[] ToArray()
        {
            ByteQue que = new ByteQue();
            que.PushSize(buff.Len);
            que.AddAll(buff.ToArray(), 0, -1);
            return que.ToArray();
        }
    }
}

File: lrpc-java/src/com/lrpc/buf/SendData.java
package com.lrpc.buf;

import com.lrpc.val.ByteQue;

/**
 * add length to the actual data to judge the integrity of the data
 */
public class SendData {

    private ByteQue buff;

    /**
     * @param que the actual data
     */
    public SendData(ByteQue que) {
        buff = que;
    }

    /**
     * added length data
     */
    public byte[] toArray() {
        ByteQue que = new ByteQue();
        que.pushSize(buff.len());
        que.addAll(buff.toArray(), 0, -1);
        return que.toArray();
    }
}